### PR TITLE
#10 resolve typescript errors

### LINKED ISSUE #10: Resolve typescript errors
Need to add definition files to the shared modules and fix other typescript errors.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #10 
Can now compile ts - have added the `yarn compile` script as well, so that we can easily generate typescript definitions.

One irritation is that eslint won't recognise the `Component :React.FC<ComponentProps>` syntax, so I've coded those manually instead.

Essentially a tidy to remove typescript errors and define types where I think required.

### CODE REVIEW

**Comment by joshxg** [comment]
Needed to make a couple updates to compile: fc84697

- styles are already exported from `*` in `./components`
- `index.ts` over `index.js` as `common` entry
- `LoadingApp` getting styles from components instead

File: host/types/bootstrap.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export {};
```
Comment (joshxg) [inline host/types/bootstrap.d.ts:1]: I'm confused why there are so many declarations?

In my very little knowledge of declaration files, I would think there would be basically a single declaration file for each frontend for any exposed modules?

File: tsconfig.base.json [hunk 2]
```diff
@@ -1,32 +1,14 @@
...
-    "target": "es5" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', 'ES2021', or 'ESNEXT'. */,
-    "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */,
+    "esModuleInterop": true /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */,
+    "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */,
     "jsx": "react" /* Specify JSX code generation: 'preserve', 'react-native', 'react', 'react-jsx' or 'react-jsxdev'. */,
-    // "declaration": true,                         /* Generates corresponding '.d.ts' file. */
-    // "declarationMap": true,                      /* Generates a sourcemap for each corresponding '.d.ts' file. */
-    "sourceMap": true /* Generates corresponding '.map' file. */,
+    "lib": ["es2018"],
+    "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */,
```
Comment (joshxg) [inline tsconfig.base.json:7]: I kinda like having all the options in the file so I don't have to google things heh

**Review by joshxg — APPROVED** [review]
Compiles for me! I need to do some learning about declaration files!

File: host/types/bootstrap.d.ts (hunk 1 above)
Comment (mark-prins) [reply host/types/bootstrap.d.ts:1]: yeah - these were generated by `tsc` and it went waaaay overboard.
I've been removing some of the extras in other branches, but I'll untangle that and tidy up this branch properly.

File: tsconfig.base.json (hunk 2 above)
Comment (mark-prins) [reply tsconfig.base.json:7]: yeah, me too. I'll pop them back in.

File: host/types/bootstrap.d.ts (hunk 1 above)
Comment (joshxg) [reply host/types/bootstrap.d.ts:1]: Ohh they were autogenerated?

I am not sure we need to 'tidy it up  properly'- I'm happy if we do it incrementally as we move forward

Comment (mark-prins) [reply host/types/bootstrap.d.ts:1]: ok, sounds good to me.

File: host/src/Host.d.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/// <reference types="react" />
+declare const Host: () => JSX.Element;
+export default Host;
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply host/src/Host.d.ts:3]: the linter was not handling it. I'll have a look and see if there's a fix, I'm already sick of changing `FC<props>` over..

**Comment by joshxg** [comment]
Gonna merge this as it touches a lot and dont want to conflict

**Pull request merged**